aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -173,6 +173,59 @@ TORCH_META_FUNC(scatter_add)
   scatter_meta_impl(*this, self, dim, index, src, "add");
 }
 
+TORCH_PRECOMPUTE_META_FUNC(index_copy)
+(const Tensor& self, int64_t dim, const Tensor& index, const Tensor& source) {
+  dim = maybe_wrap_dim(dim, self.dim());
+
+  const Tensor& result = maybe_get_output(0);
+  bool check_result = result.defined();
+  set_output(self.sizes(), self.options());
+  if (check_result) {
+    at::assert_no_internal_overlap(result);
+    at::assert_no_overlap(result, index);
+    at::assert_no_overlap(result, source);
+  }
```

Comment:
Note to self: this needs a comment. (https://github.com/pytorch/pytorch/pull/63312/files#diff-c4e35e76279419d9edda979a995412ee494fad3a97d3fb59486a1f0e326d48beR91-R95)